src/app/layout/qgslayoutlegendwidget.cpp

```diff
@@ -900,6 +903,44 @@ void QgsLayoutLegendWidget::mExpressionFilterButton_toggled( bool checked )
   mLegend->endCommand();
 }
 
+void QgsLayoutLegendWidget::mLayerExpressionButton_clicked()
```

Comment:
This is a new GUI function located below the legend in the Item Property, the goal of the button is to convert the current label to the symbol label and allow the user to insert other expressions.

This expression should be applied to all nested nodes of the layer.